plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -75,8 +81,17 @@ public static void parsePartitionField(PartitionSpec.Builder builder, String fie
         }
     }
 
-    private static String fromIdentifier(String identifier)
+    // Currently, all Iceberg columns are stored in lowercase in the Iceberg metadata files.
+    // Unquoted identifiers are canonicalized to lowercase here which is not according ANSI SQL spec.
+    // Quoted identifiers are restricted to lowercase only through the regex pattern.
```

Comment:
the `[^A-Z]` isn't sufficient for that.
What about `Ą`?

simplify regex, and use `.toLowerCase(ENGLISH)` in Java to verify parsed value is all-lower for now